entry/src/block_component.rs

```diff
@@ -699,7 +749,15 @@ impl VersionedBlockMarker {
         };
 
         // Total size includes the version field
-        Ok(VERSION_SIZE + marker_inner_size)
+        Ok(Self::VERSION_SIZE + marker_inner_size)
+    }
+
+    /// Returns the serialized size in bytes without actually serializing.
+    fn serialized_size(&self) -> Result<u64, BlockComponentError> {
```

Comment:
this can be updated to not return a `Result` if the functions it is calling are also updated.